dll/win32/shell32/CShellLink.cpp

```diff
@@ -2592,40 +2592,33 @@ HRESULT STDMETHODCALLTYPE CShellLink::InvokeCommand(LPCMINVOKECOMMANDINFO lpici)
 
 HRESULT CShellLink::DoOpen(LPCMINVOKECOMMANDINFO lpici)
 {
-    HRESULT hr;
-    LPWSTR args = NULL;
-    LPWSTR path = strdupW(m_sPath);
     BOOL unicode = lpici->cbSize >= FIELD_OFFSET(CMINVOKECOMMANDINFOEX, ptInvoke) &&
                    (lpici->fMask & CMIC_MASK_UNICODE);
 
+    CStringW args;
+    if (m_sArgs)
+        args = m_sArgs;
+
+    CComHeapPtr<WCHAR> pszParams;
     if (unicode)
     {
         LPCMINVOKECOMMANDINFOEX iciex = (LPCMINVOKECOMMANDINFOEX)lpici;
-        SIZE_T len = 2;
-
-        if (m_sArgs)
-            len += wcslen(m_sArgs);
-        if (iciex->lpParametersW)
-            len += wcslen(iciex->lpParametersW);
-
-        args = (LPWSTR)HeapAlloc(GetProcessHeap(), 0, len * sizeof(WCHAR));
-        *args = 0;
-        if (m_sArgs)
-            wcscat(args, m_sArgs);
         if (iciex->lpParametersW)
```

Comment:
```suggestion
        if (iciex->lpParametersW && iciex->lpParametersW[0])
```
see https://github.com/wine-mirror/wine/commit/14490909856e6bea715e215625d7461082a146d1